crypto/evp/evp_pkey.c

```diff
@@ -73,6 +73,18 @@ EVP_PKEY *EVP_PKCS82PKEY_ex(const PKCS8_PRIV_KEY_INFO *p8, OSSL_LIB_CTX *libctx,
     int selection;
     size_t len;
     OSSL_DECODER_CTX *dctx = NULL;
+    const ASN1_OBJECT *algoid = NULL;
+    const char *keytype = NULL;
+    char keytypebuf[80];
+
+    if (PKCS8_pkey_get0(&algoid, NULL, NULL, NULL, p8)) {
+        if (OBJ_obj2txt(keytypebuf, sizeof(keytypebuf), algoid, 0))
+            keytype = keytypebuf;
+    }
+    /*
+     * else we don't know the OID...carry on anyway to see if we may have a
+     * decoder that can handle it anyway
+     */
```

Comment:
Hmm....actually....does the subject of this comment ever reasonably happen now with this latest change? We should always get an OID out...otherwise it's not PKCS8

Reply:
Heh, I came back here 'cause I thought of exactly this.  So, drop the comment, I guess?